src/filesystem_helper.cpp

```diff
@@ -306,6 +306,22 @@ path temp_directory_path()
   return path(temp_path);
 }
 
+path create_temp_directory(const std::string & base_name, const path & parent)
+{
+  auto template_path = base_name + ".XXXXXX";
+
+#ifdef _WIN32
+  _mktemp_s(&template_path[0], template_path.size() + 1);
+  const auto final_path = parent / template;
+  create_directories(final_path);
+  return final_path
+#else
+  auto full_template = (parent / template_path).string();
+  const char * dir_name = mkdtemp(&full_template[0]);
```

Comment:
i think we need to check pointer if it is NULL or not, and we cannot guarantee this function succeeds always since system call would fail based on user input.

Reply:
Also, consider using [`std::string::c_str()`](https://en.cppreference.com/w/cpp/string/basic_string/c_str) instead.